integrations/api_issue_subscription_test.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package integrations
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"code.gitea.io/gitea/models"
+	api "code.gitea.io/gitea/modules/structs"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAPIIssueSubscriptions(t *testing.T) {
```

Comment:
These sorts of tests are really just unit tests - they're not really integration - these simply test whether we get the answer back given the correct data in the first place. They're useful to have but can be beyond irritating to fix when they break or become out of date.

We need more story tests - similar to git_test.go - ones where a user creates/forks/mirrors repos, adds issues, collaborators, teams etc makes pull requests and pushes, checks issues close automatically, that notifications are fired appropriately, that users are/are not subscribed etc. These sort of tests actually test how a user interacts with the system and can be used to interrogate how data changes in multiple places through testing multiple things in after each action and through the use of test components.

Reply:
I only can call http urls and check if it matches with fixtures in integration - shuld I try to move it somehow?